aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+.. literalinclude:: razor/sample/Views/Home/Contact10.cshtml
+  :language: html
+
+The ``@inherits`` keyword is not allowed on the same page with the ``@model`` statement. You can pass a model as shown with the following Razor page (when passed "Rick@contoso.com" in the model):
+
+.. literalinclude:: razor/sample/Views/Home/Login1.cshtml
+  :language: html
+  :lines: 1-
+
+Generates this HTML:
+
+.. code-block:: none
+
+  <div>The Login Email: Rick@contoso.com</div>
+  <div>Custom text: Hello World.</div>
+
+.. review: Adding the model to _ViewImports is not needed. We don't need it to pass a model.
+
+While you can't use ``@model`` and ``@inherits`` on the same page, you can have ``@model`` in a *_ViewImports.cshtml* file that the Razor page imports. See :doc:`/mvc/views/layout`. For example, if your Razor view imported the following *_ViewImports.cshtml* file:
+
+.. literalinclude:: razor/sample/Views/_ViewImportsModel.cshtml
+  :language: html
+
+The following Razor page, when passed "Rick@contoso.com" in the model:
+
+.. literalinclude:: razor/sample/Views/Home/Login2.cshtml
+  :language: html
+
+Generates this HTML markup:
+
+.. code-block:: none
+
+  <div>The Login Email: Rick@contoso.com</div>
+  <div>Custom text: Hello World</div>
+
+
+``@inject``
+^^^^^^^^^^^^^^
+The ``@inject`` directive enables you to inject a service from your :doc:`service container </fundamentals/dependency-injection>`  into your Razor page for use. See :doc:`/mvc/views/dependency-injection`.
+
+Much like ``@inherits`` you can also provide the ``TModel`` parameter if your service happens to depend on the current model type:
+
+.. code-block:: none
+
+  @inject IHtmlHelper<TModel> CustomHtmlHelper
+
+  @CustomHtmlHelper.Raw("<div>Hello World</div>")
+
+An additional feature of ``@inject`` is that it also enables you to replace a few properties that are automatically injected into a Razor page. For instance, we could replace the Html property on our Razor page with the hosting environment:
+
+.. literalinclude:: razor/sample/Views/Home/Contact4.cshtml
+  :language: html
+
+If you inject a property that already exists on your Razor page and has been ``@injected`` before, or is one of the following properties:
+
+- Html
+- Json
+- Component
+- Url
+- :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.ModelExpressionProvider`
+
+Your ``@inject`` statement will override that property and it will be available as the overridden type.
+
+``@functions``
+^^^^^^^^^^^^^^
+
+The ``@functions`` directive enables you to add function level content to your Razor page. The syntax is:
+
+.. code-block:: none
+
+  @functions { // C# Code }
+
+For example:
+
+.. literalinclude:: razor/sample/Views/Home/Contact6.cshtml
+  :language: html
+
+Generates the following HTML markup:
+
+.. code-block:: none
+
+  <div>From method: Hello</div>
+
+The generated Razor C# looks like:
+
+.. literalinclude:: razor/sample/Classes/Views_Home_Test_cshtml.cs
+  :language: c#
+  :lines: 1-19
+
+``@section``
```

Comment:
The `@section` directive is already covered in the [Layout](https://docs.asp.net/en/latest/mvc/views/layout.html#id1). Can we reference that instead of duplicating the content here?